phac-nml/irida
# Object store/ connection check on startup

## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Updated startup checks for cloud providers to make sure a connection is made to the bucket/container otherwise the startup is exited. Also, updated the filesystem startup to check if a basedirectory is writeable. If not then startup is exited as well.

**To test cloud connection:**
1) In the `irida.conf` file change the bucket name for aws or the container name for azure. Also, change the `irida.storage.type` to either aws or azure
2) Startup IRIDA
3) IRIDA should not startup and there should be an exception in the console advising you to check the credentials and that the bucket/container exist

**To test local file system directory writable check:**
1) Make one of the output directories in `irida.conf` non writable
2) Startup IRIDA
3) IRIDA should not startup and there should be an exception in the console advising you that the directory is not writable and to check the directory permissions

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
~* [ ] Tests added (or description of how to test) for any new features.~
~* [ ] User documentation updated for UI or technical changes.~
